actor/src/main/scala-3/org/apache/pekko/util/ByteString.scala

```diff
@@ -825,6 +876,32 @@ sealed abstract class ByteString
   // optimized in subclasses
   override def indexOf[B >: Byte](elem: B, from: Int): Int = indexOf(elem, from)
 
+  // optimized version of indexOf for bytes, optimized in subclasses
+  /**
+   * Finds index of first occurrence of some byte in this ByteString after or at some start index.
+   *
+   * Similar to indexOf, but it avoids boxing if the value is already a byte.
+   *
+   *  @param   elem   the element value to search for.
+   *  @param   from   the start index
+   *  @return  the index `>= from` of the first element of this ByteString that is equal (as determined by `==`)
+   *           to `elem`, or `-1`, if none exists.
+   *  @since 1.1.0
+   */
+  def indexOf(elem: Byte, from: Int): Int = indexOf(elem, from)
```

Comment:
Infinite loop here (probably impossible to be called)?

Reply:
Replaced with a `???` should be better I think.

Reply:
Can just be left blank, no need for a default implementation afaics.

Reply:
Removed default implementation in 2cce3ea

I think this will fallback to `indexWhere`